fedbiomed/researcher/job.py

```diff
@@ -174,6 +178,21 @@ def __init__(self,
         self.validate_minimal_arguments(self._repository_args,
                                         ['training_plan_url', 'training_plan_class', 'params_url'])
 
+    def _load_training_plan_from_file(self, training_plan_module: str) -> BaseTrainingPlan:
+        """Import a training plan class from a file and create a training plan object instance.
+
+        Args:
+            training_plan_module: module name of the training plan file
+
+        Returns:
+            The training plan object created
+        """
+        sys.path.insert(0, self._keep_files_dir)
+        module = importlib.import_module(training_plan_module)
```

Comment:
Rather than compute `os.path.split(self._training_plan_file)[1][:-3]` (makes assumption on file name structure, thus difficult to maintain) I would propose to pass a training_plan_module variable.

=> I push a patch for this